### PR TITLE
Add new lecture material on tidy data and data entry

### PR DESCRIPTION
**Pull request by ethanwhite**

### CODE REVIEW

File: materials/data-entry.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+### Where to enter data
+
+* Spreadsheet
+* Text file
+* Database
+* Form (web or GUI databases)
+
+* All of these are reasonable options
+* When used properly spreadsheets or forms for databases can provide additional
+project against bad data being entered.
```
Comment (brymz) [inline materials/data-entry.md]: 'project' -> 'protection' ?

File: materials/data-entry.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+* Be careful of data conversion issues from spreadsheets
+
+> Show posts/papers on dates and gene names
```
Comment (brymz) [inline materials/data-entry.md]: links?

File: materials/data-entry.md (hunk 1 above)
Comment (ethanwhite) [inline materials/data-entry.md]: "protect" actually. Fixed.

File: materials/data-entry.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+---
+layout: page
+element: notes
+title: Data Entry
+language: SQL
```
Comment (brymz) [inline materials/data-entry.md]: Excel?

File: materials/data-entry.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+### Where to enter data
+
+* Spreadsheet
+* Text file
+* Database
+* Form (web or GUI databases)
+
+* All of these are reasonable options
+* When used properly spreadsheets or forms for databases can provide additional
+protect against bad data being entered
```
Comment (brymz) [inline materials/data-entry.md]: 'protection'

File: materials/tidy-data.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+title: Tidy Data
+language: SQL
+---
+
+The main goal of tidy data is making it easy for a computer to work with the
+data. Let's start by looking at some messy data and thinking about what makes it
+messy and what we could do to improve it.
+
+> Do the exercise on [Improving Messy Data]({{ site.baseurl }}/exercises/Tidy-data-improving-messy-data-SQL).
+
```
Comment (brymz) [inline materials/tidy-data.md]: >

File: materials/data-entry.md (hunk 2 above)
Comment (ethanwhite) [inline materials/data-entry.md]: Good call. Added.

File: materials/data-entry.md (hunk 3 above)
Comment (ethanwhite) [inline materials/data-entry.md]: Wasn't sure is this would break something else. Done.

File: materials/data-entry.md (hunk 4 above)
Comment (ethanwhite) [inline materials/data-entry.md]: why yes, yes it is. sorry about that.

File: materials/tidy-data.md (hunk 5 above)
Comment (ethanwhite) [inline materials/tidy-data.md]: I don't understand.

File: materials/data-entry.md (hunk 3 above)
Comment (brymz) [inline materials/data-entry.md]: The `notes` are free of `language` constraints.

File: materials/tidy-data.md (hunk 5 above)
Comment (brymz) [inline materials/tidy-data.md]: A super minor point...

You have a blank line in between the `> Do action` and the points that you hope to cover during that action. I place a `>` in that blank line to keep the block quote bar continuous (one action).

Comment (ethanwhite) [inline materials/tidy-data.md]: Got it. Fixed.

**Comment by brymz** [comment]
A broad comment... Any of the `materials/` that you have added will have to be added manually to `materials/index`.

It is manual because of the `IPYNB` and `PDF` formats without `yaml` structure, though we could consider adding some sort of automation or check.

**Pull request merged**